packages/account/src/providers/provider.test.ts

```diff
@@ -1039,6 +1040,76 @@ describe('Provider', () => {
     await Promise.all(promises);
   });
 
+  it('should not throw if the subscription stream data string contains more than one "data:"', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    vi.spyOn(global, 'fetch').mockImplementationOnce(() => {
+      const responseObject = {
+        data: {
+          submitAndAwait: {
+            type: 'SuccessStatus',
+            time: 'data: 4611686020137152060',
+          },
+        },
+      };
+      const streamedResponse = new TextEncoder().encode(
+        `data:${JSON.stringify(responseObject)}\n\n`
+      );
+      return Promise.resolve(
+        new Response(
+          new ReadableStream({
+            start: (controller) => {
+              controller.enqueue(streamedResponse);
+              controller.close();
+            },
+          })
+        )
+      );
+    });
+
+    for await (const { submitAndAwait } of provider.operations.submitAndAwait({
+      encodedTransaction: "it's mocked so doesn't matter",
+    })) {
+      expect(submitAndAwait.type).toEqual('SuccessStatus');
+      expect((<SubmittedStatus>submitAndAwait).time).toEqual('data: 4611686020137152060');
+    }
+  });
+
+  it('should throw if the subscription stream data string parsing fails for some reason', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    const badResponse = 'data: whatever';
+    vi.spyOn(global, 'fetch').mockImplementationOnce(() => {
+      const streamResponse = new TextEncoder().encode(badResponse);
+      return Promise.resolve(
+        new Response(
+          new ReadableStream({
+            start: (controller) => {
+              controller.enqueue(streamResponse);
+              controller.close();
+            },
+          })
+        )
+      );
+    });
+
+    await expectToThrowFuelError(
+      async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        for await (const { submitAndAwait } of provider.operations.submitAndAwait({
```

Comment:
Nit.

```suggestion
        for await (const { submitAndAwait: _ } of provider.operations.submitAndAwait({
```

Reply:
It makes use of [this](https://github.com/FuelLabs/fuels-ts/blob/950f9be3856bbd00ee11c1d41f38729e0ea9c653/.eslintrc.js#L63-L68) rule exception.

Reply:
TS didn't accept that 😢